pkg/lightning/config/config.go

```diff
@@ -1381,45 +1381,30 @@ func (c *Conflict) adjust(i *TikvImporter, l *Lightning) error {
 		case ErrorOnDup:
 			c.Threshold = 0
 		case IgnoreOnDup, ReplaceOnDup:
-			c.Threshold = math.MaxInt64
+			c.Threshold = DefaultRecordDuplicateThreshold
 		case NoneOnDup:
 			c.Threshold = 0
-			if i.Backend == BackendLocal && c.Strategy != NoneOnDup {
-				c.Threshold = math.MaxInt64
-			}
 		}
 	}
 	if c.Threshold > 0 && c.Strategy == ErrorOnDup {
 		return common.ErrInvalidConfig.GenWithStack(
 			`conflict.threshold cannot be set when use conflict.strategy = "error"`)
 	}
 
-	if c.MaxRecordRows < 0 {
-		maxErr := l.MaxError
-		// Compatible with the old behavior that records all syntax,charset,type errors.
-		maxAccepted := max(maxErr.Syntax.Load(), maxErr.Charset.Load(), maxErr.Type.Load())
-		if maxAccepted < defaultMaxRecordRows {
-			maxAccepted = defaultMaxRecordRows
-		}
-		if maxAccepted > c.Threshold {
-			maxAccepted = c.Threshold
-		}
-		if c.Strategy == ReplaceOnDup && i.Backend == BackendTiDB {
-			// due to we use batch insert, we can't know which row is duplicated.
-			maxAccepted = 0
+	if c.Strategy == ReplaceOnDup && i.Backend == BackendTiDB {
+		// due to we use batch insert, we can't know which row is duplicated.
+		if c.MaxRecordRows >= 0 {
+			// only warn when it is set by user.
+			log.L().Warn(`Cannot record duplication (conflict.max-record-rows > 0) when use tikv-importer.backend = \"tidb\" and conflict.strategy = \"replace\".
```

Comment:
this behaviour also changed. In the old behaviour tidb backend + replace will see error if set max-record-rows, the error will tell the user can't find the error details in conflict table in advance. Now the user will not see the error. Please double check with PM.

Reply:
PM confirmed that the user will no longer see this error. Instead there will be a warning in the log